pkg/utils/kubeconfig/kubeconfig.go

```diff
@@ -99,10 +99,24 @@ func AppendAuthenticator(config *clientcmdapi.Config, spec *api.ClusterConfig, a
 		roleARNFlag string
 	)
 
+	execConfig := &clientcmdapi.ExecConfig{
+		APIVersion: "client.authentication.k8s.io/v1alpha1",
+		Command:    authenticatorCMD,
+	}
+
 	switch authenticatorCMD {
 	case AWSIAMAuthenticator, HeptioAuthenticatorAWS:
 		args = []string{"token", "-i", spec.Metadata.Name}
 		roleARNFlag = "-r"
+		if spec.Metadata.Region != "" {
```

Comment:
Is it worth adding tests for that?
What would break if there were to be a regression there?

Reply:
We do have tests for this, but not one with the region unset. It's a small check and a lot of our tests rely on the region being present.